lisa/schema.py

```diff
@@ -353,88 +355,355 @@ def __post_init__(self, *args: Any, **kwargs: Any) -> None:
             )
 
 
+FEATURE_NAME_RDMA = "RDMA"
+
+
 @dataclass_json(letter_case=LetterCase.CAMEL)
 @dataclass
-class IntegerRange:
-    min: int
-    max: int
+class Feature:
+    name: str = ""
+    enabled: bool = True
+    can_disable: bool = False
+
+
+class Features:
+    RDMA = Feature(name=FEATURE_NAME_RDMA)
 
 
 @dataclass_json(letter_case=LetterCase.CAMEL)
 @dataclass
-class NodeSpec(ExtendableSchemaMixin):
+class NodeSpace(search_space.RequirementMixin, ExtendableSchemaMixin):
```

Comment:
Hm, at some point I want to review what Kubernetes and Mesos did for node features and review this more thoroughly. It's certainly not something we need to reinvent (though obviously we need to write it in such a way that it works for us, in our codebase). They both generally refer to them as "resources." Here are the [Kubernetes docs](https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/) and the [Mesos docs](http://mesos.apache.org/documentation/latest/attributes-resources/). I'd recommend reading them just to get a feel for the design space here (we're doing essentially the same thing!). I _also_ need to re-read these, as it's been a while.

Reply:
Ya, I'm not sure. This and the next file I feel like take more code than necessary. We should have a generic way of creating requirements, such that we can just have a list of features and requirements and match them. But it's also just a feeling. Mesos certainly had a lot of code dealing with node requirements too.

Reply:
Thank you for pointing out it, I haven't found similar things before. I didn't implement it well, I will try to use this part more clear, so we have chance to refactor it in future.